packages/react-spinbutton/Spec.md

```diff
@@ -2,62 +2,381 @@
+| step      | The granualarity of the value when incrementing or decrementing                 |
+
+Despite supporting both `min` and `max` attributes a native number input will allow users to enter values outside the specified bounds. This situation is resolved via a process called [constraint validation](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Constraint_validation) that adds CSS pseudo classes to the element for styling purposes and raises validation events.
+
+There are very few options for cross-browser styling of native number inputs. The [::-webkit-inner-spin-button](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-inner-spin-button) pseudo element allows for selecting the spin buttons of a number input but is only supported by Webkit and Blink based browsers.
+
+Native number inputs are meant strictly for number input but what constitutes number input is inconsistent across browsers ([see this Bugzilla issue for details](https://bugzilla.mozilla.org/show_bug.cgi?id=1398528)). You can easily see this on [MDN's simple example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#a_simple_number_input). In Edge 96 you can enter exponential numbers like "1e+343434" but not arbitrary strings like "cats". On the same example in Firefox 95 you can enter both "1e+343434" and "cats".
+
+Inspecting a native number input with devtools shows that it implements the [spinbutton ARIA attributes as described by WAI-ARIA](https://www.w3.org/TR/wai-aria-practices/#spinbutton)
+
+#### Examples
+
+- [Controlling Step Size](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#controlling_step_size)
+- [Specifying Minimum and Maximum Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#specifying_minimum_and_maximum_values)
+- [Allowing Decimal Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#allowing_decimal_values)
+- [Offering Suggested Values](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number#offering_suggested_values)
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+### Uncontrolled Example
+
+```tsx
+<SpinButton label="Uncontrolled Example" defaultValue="1" />
+```
+
+### Controlled Example
+
+```tsx
+type SpinButtonChangeData = {
+  value: number; // The "raw value", e.g., 2
+  formattedValue: string; // The "formatted value", e.g., "$2.00"
+  prevValue: number;
+  prevFormattedValue: string;
+};
+
+const [value, setValue] = useState<number>(2);
+const onControlledExampleChange = (_event, data: SpinButtonChangeData) => {
+  setValue(data.value);
+};
+
+<SpinButton label="Controlled Example" value={value} onChange={onControlledExampleChange} />;
+```
+
+### Custom Display Format Example
+
+```tsx
+type SpinButtonChangeData = {
+  value: number;
+  formattedValue: string;
```

Comment:
I would leave the formatted value out of the change handler payload, if the user is already passing in a formatter then they can use it on the value.

It also reduces the API surface to only deal with numerical values